Theme overrides for the MUI Chip for Pills and Badges components

iandilling-adaptive opened the pull request:
There's so much overlap between the Pills and Button components, that these theme overrides can likely be shared between the two. The Button could be refactored to set the overrides, and then the MUI Button can be used for many of the Button and Pills use cases.

tprince232 commented on docs/src/components/mui/Chip.stories.tsx:
```diff
@@ -0,0 +1,80 @@
+);
+
+export default {
+  title: 'Material UI/Chip',
```
Might want to add `(Pill)` or to the title so show somewhere that chip === pill

iandilling-adaptive replied:
I think I'm just going to make a Pill component in our components package and within it make use of the Chip component and it's style overrides. There is a special case for stock tickers that it could implement, plus the left/right icon semantics can be improved: for instance Chip calls the right icon the deleteIcon, but that icon might not always be used for deletion in our design system.

tprince232 replied:
Makes sense.

So you'll leave this here as "Chip" from MUI and then you'll have a new PR building on this for our Pill?

iandilling-adaptive replied:
Yeah. I can of course also make this `Material UI/Chip (Pill)` just so it's clear what this can be used for in the design system.

tprince232 commented:
One minor comment from me - looks complex w/ lots of overrides for various states but also straightforward - overriding each scenario as needed.

Regarding your comment about sharing the theme overrides between Pill & Button - do you think that's something worth doing as this PR / is there a reason you left it out for now / should you make a work item to track that?

wenfeng-xing commented on docs/src/components/mui/Button.stories.tsx:
```diff
@@ -0,0 +1,61 @@
+import { Button as MuiButton, ButtonProps } from '@mui/material';
```
Maybe for the chip, we really need to override is the Chip component in MUI https://mui.com/material-ui/react-chip/

iandilling-adaptive replied:
I added this story because I was checking the MUI button to see how it behaved when I changed the base style overrides. I am going to leave this story as it can be useful to see how the basic MUI button is rendered in the theme, but I will remove some of the overrides causing issues in the existing button.

iandilling-adaptive commented:
> One minor comment from me - looks complex w/ lots of overrides for various states but also straightforward - overriding each scenario as needed.
> 
> Regarding your comment about sharing the theme overrides between Pill & Button - do you think that's something worth doing as this PR / is there a reason you left it out for now / should you make a work item to track that?

I didn't want to replace the existing implementation and accidentally break something, plus it seemed out of scope for this change. I actually see now that the buttons have a rounder border radius, only use primary colors, and have slightly different interact states, so they aren't capable of being full replacements for clickable pills unless we add some extra variant. I do think that at least the outline can be shared, which I added as part of MuiButtonBase (Chips inherit this when they becoming clickable).

I think for now I'll leave the button as is, remove the default button overrides which I think is breaking the current button implementation, and leave the MuiButtonBase override for when/if we want to update the buttons to use it.

iandilling-adaptive commented:
Latest changes: Removed some of the default MuiButton overrides, kept the MuiButtonBase override, wrapped MuiChip in a Pill component to extend its functionality, and added a story for the Pill component.

iandilling-adaptive commented:
Latest change: Set the focus outline color to the correct value for dark mode.
